krm-functions/lib/kptfile/v1/kpt_test.go

```diff
@@ -0,0 +1,190 @@
+				{Type: "c", Status: kptv1.ConditionFalse, Reason: "c", Message: "c"},
+				{Type: "a", Status: kptv1.ConditionFalse, Reason: "a", Message: "a"},
+			},
+			want: []kptv1.Condition{
+				{Type: "b", Status: kptv1.ConditionFalse, Reason: "b", Message: "b"},
+				{Type: "c", Status: kptv1.ConditionFalse, Reason: "c", Message: "c"},
+				{Type: "a", Status: kptv1.ConditionFalse, Reason: "a", Message: "a"},
+			},
+		},
+		"SetConditionsNonEmpty": {
+			cs: []kptv1.Condition{
+				{Type: "x", Status: kptv1.ConditionFalse, Reason: "x", Message: "x"},
+				{Type: "y", Status: kptv1.ConditionFalse, Reason: "y", Message: "y"},
+			},
+			t: []kptv1.Condition{
+				{Type: "b", Status: kptv1.ConditionFalse, Reason: "b", Message: "b"},
+				{Type: "c", Status: kptv1.ConditionFalse, Reason: "c", Message: "c"},
+				{Type: "a", Status: kptv1.ConditionFalse, Reason: "a", Message: "a"},
+			},
+			want: []kptv1.Condition{
+				{Type: "x", Status: kptv1.ConditionFalse, Reason: "x", Message: "x"},
+				{Type: "y", Status: kptv1.ConditionFalse, Reason: "y", Message: "y"},
+				{Type: "b", Status: kptv1.ConditionFalse, Reason: "b", Message: "b"},
+				{Type: "c", Status: kptv1.ConditionFalse, Reason: "c", Message: "c"},
+				{Type: "a", Status: kptv1.ConditionFalse, Reason: "a", Message: "a"},
+			},
+		},
+		"SetConditionsOverlap": {
+			cs: []kptv1.Condition{
+				{Type: "x", Status: kptv1.ConditionFalse, Reason: "x", Message: "x"},
+				{Type: "y", Status: kptv1.ConditionFalse, Reason: "y", Message: "y"},
+			},
+			t: []kptv1.Condition{
+				{Type: "b", Status: kptv1.ConditionFalse, Reason: "b", Message: "b"},
+				{Type: "y", Status: kptv1.ConditionFalse, Reason: "ynew", Message: "ynew"},
+				{Type: "a", Status: kptv1.ConditionFalse, Reason: "a", Message: "a"},
+			},
+			want: []kptv1.Condition{
+				{Type: "x", Status: kptv1.ConditionFalse, Reason: "x", Message: "x"},
+				{Type: "y", Status: kptv1.ConditionFalse, Reason: "ynew", Message: "ynew"},
+				{Type: "b", Status: kptv1.ConditionFalse, Reason: "b", Message: "b"},
+				{Type: "a", Status: kptv1.ConditionFalse, Reason: "a", Message: "a"},
+			},
+		},
+	}
+
+	for name, tc := range cases {
+		kf := NewMutator(f)
+		if _, err := kf.UnMarshal(); err != nil {
+			t.Errorf("cannot unmarshal file: %s", err.Error())
+		}
+		t.Run(name, func(t *testing.T) {
+			if tc.cs != nil {
+				kf.SetConditions(tc.cs...)
+			}
+			if tc.t != nil {
+				kf.SetConditions(tc.t...)
+			}
+			gots := kf.GetConditions()
+			if len(gots) != len(tc.want) {
+				t.Errorf("TestSetConditions: got: %v, want: %v", gots, tc.want)
+			} else {
+				for idx, got := range gots {
+					// no need to validate length as this was already done
+					/*
+						if idx > len(tc.want)-1 {
+							t.Errorf("TestSetConditions: got: %v, want: %v", gots, tc.want)
+						}
+					*/
+					if diff := cmp.Diff(tc.want[idx], got); diff != "" {
+						t.Errorf("TestGetCondition: -want, +got:\n%s", diff)
+					}
+				}
+			}
+		})
+	}
+}
```

Comment:
Tests are missing for other methods such as DeleteCondition, Unmarshall etc.

Reply:
added the tests for these. DeleteCondition (here you see if I delete the last one it does not matter, so the procedure with append is safe).

80% coverage now